account-kit/signer/src/base.ts

```diff
@@ -809,6 +817,74 @@ export abstract class BaseAlchemySigner<TClient extends BaseSignerClient>
     });
   };
 
+  /**
+   * This method lets you adapt your AlchemySigner to a viem WalletClient.
+   *
+   * @param {ToViemWalletClientParams} opts - options
+   * @example
+   * ```ts
+   * import { AlchemyWebSigner } from "@account-kit/signer";
+   *
+   * const signer = new AlchemyWebSigner({
+   *  client: {
+   *    connection: {
+   *      rpcUrl: "/api/rpc",
+   *    },
+   *    iframeConfig: {
+   *      iframeContainerId: "alchemy-signer-iframe-container",
+   *    },
+   *  },
+   * });
+   *
+   * const client = signer.toViemWalletClient();
+   * ```
+   *
+   * @throws if your signer is not authenticated
+   * @returns {WalletClient} a viem WalletClient object that can be used with viem's wallet client
+   */
+  toViemWalletClient = <
+    TTransport extends Transport,
+    TChain extends Chain | undefined = undefined,
+    TRpcSchema extends RpcSchema | undefined = undefined
+  >({
+    transport,
+    ...opts
+  }: Omit<
+    WalletClientConfig<TTransport, TChain, LocalAccount, TRpcSchema>,
+    "account"
+  >): WalletClient<TTransport, TChain, LocalAccount, TRpcSchema> => {
+    const viemAccount = this.toViemAccount();
+
+    // Construct fallback transport: https://github.com/wevm/viem/blob/59d2c093fe435e97684d34266f72c432dd6a0b3a/src/clients/createClient.ts#L231-L234
+    const fallbackTransport = transport({
+      chain: opts.chain,
+      pollingInterval: opts.pollingInterval ?? 4_000,
+    });
+
+    return createWalletClient({
+      ...opts,
+      account: viemAccount,
+      transport: custom({
+        async request({ method, params }) {
+          switch (method) {
+            case "personal_sign":
+              return viemAccount.signMessage(
+                isHex(params[0])
+                  ? { message: { raw: params[0] } }
+                  : { message: params[0] }
+              );
+            case "eth_signTypedData_v4":
+              return viemAccount.signTypedData(params[1]);
```

Comment:
Should we assert `params[0]` to be the account address?

Reply:
Yea +1

Reply:
And probs same above checking that params[1] is the signer too

Reply:
`params[1]` is the typed data object, so I don't think there's anything we can assert over that.